sqlx-macros-core/src/query/data.rs

```diff
@@ -152,21 +152,23 @@ where
 
     pub(super) fn save_in(&self, dir: impl AsRef<Path>) -> crate::Result<()> {
         let path = dir.as_ref().join(format!("query-{}.json", self.hash));
-        let mut file = atomic_write_file::AtomicWriteFile::open(&path)
-            .map_err(|err| format!("failed to open the temporary file: {err:?}"))?;
-
-        serde_json::to_writer_pretty(file.as_file_mut(), self)
-            .map_err(|err| format!("failed to serialize query data to file: {err:?}"))?;
+        std::fs::remove_file(&path).map_err(|err| format!("failed to delete {path:?}: {err:?}"))?;
```

Comment:
Breaks if the file doesn't exist already, which would be the case on a first run or if a concurrent invocation deleted the file already.